core/runtime/execute_engine.cpp

```diff
@@ -168,27 +210,66 @@ std::vector<at::Tensor> execute_engine(std::vector<at::Tensor> inputs, c10::intr
       auto dims = core::util::toVec(out_shape);
       auto type = util::TRTDataTypeToScalarType(compiled_engine->exec_ctx->getEngine().getTensorDataType(name.c_str()));
       outputs[pyt_idx] = std::move(at::empty(dims, {at::kCUDA}).to(type).contiguous());
+
+      // In cudagraphs mode, the allocated output buffers are stored for reuse
```

Comment:
Is there a downside to always using the buffers?

Reply:
Also what is the required lifetime of these buffers, can they be deallocated at any point or do they need to be reserved as long as the cuda graph is active?

Reply:
If not in cudagraphs mode, the main downside is the requirement that buffers need to change size/be reallocated in Dynamic shape usecases.

My understanding is that these I/O buffers need to be reserved for the duration of the cudagraph being active